Files implementation

Andrewangeta opened the pull request.

Andrewangeta committed:
Clean up.

Andrewangeta committed:
Implemented file links and file uploads support.

Andrewangeta commented:
This addresses https://github.com/vapor-community/stripe-provider/issues/66

anthonycastelli commented on Sources/Stripe/API/Routes/FileRoutes.swift:
```diff
@@ -0,0 +1,101 @@
+        // Fordm data structure found here.
+        // https://www.w3.org/TR/html401/interact/forms.html#h-17.13.4
+        
+        let boundary = "Stripe-Vapor-\(UUID().uuidString)"
```
Vapor has a Mutlipart library that you can import to take care of all this heavy lifting. I'm not sure how it all works in Vapor 3, but I know it's still around.

Andrewangeta replied:
Ah I see. I just checked it out and I'd have to have multiple structs to be able to take advantage of  it. I'd rather just hand write it. I've done this on iOS a lot too.

anthonycastelli replied:
Sounds good to me. If it becomes problematic we can look into swapping it over.

Andrewangeta committed:
Small update for upload request. Fixed header position.

vzsg committed:
Fix image uploads by allowing non-String bodies in StripeRequest

The issue boiled down to StripeRequest, where the `body` parameter was always expected to be a human-friendly String.
Real image data is typically not representable as UTF-8 strings, so the forced conversion would always fail this way or another.

The change from String to LosslessHTTPBodyRepresentable does not affect library code and should not affect users either –
any String that was used before automatically complies to this more generic protocol.
